packages/viewer/src/lib/volume-intersection/controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Point } from '@vertexvis/geometry';
+
+import { SceneItemOperationsBuilder } from '../scenes';
+import { VolumeIntersectionQueryModel } from './model';
+
+export const DEFAULT_CONCURRENT_VOLUME_QUERY_LIMIT = 5;
+
+export type OperationTransform = (
+  builder: SceneItemOperationsBuilder
+) => SceneItemOperationsBuilder;
+
+export class VolumeIntersectionQueryController {
+  private previousViewerCameraControls?: boolean;
+  private operationTransform: OperationTransform;
+  private inFlightOperations = 0;
```

Comment:
nit: this could be configurable as a controller setting.

Reply:
Yeah, my thinking was to come back and revisit this once we had more information from testing, but it's easy enough to make this configurable for now as well. I'll update this